src/main/java/org/elasticsearch/plugin/analysis/skroutzgreekstemmer/SkroutzGreekStemmerPlugin.java

```diff
@@ -1,23 +1,20 @@
 package org.elasticsearch.plugin.analysis.skroutzgreekstemmer;
 
+import org.elasticsearch.indices.analysis.AnalysisModule.AnalysisProvider;
+import org.elasticsearch.index.analysis.TokenFilterFactory;
+import org.elasticsearch.plugins.AnalysisPlugin;
 import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.index.analysis.AnalysisModule;
-import org.elasticsearch.index.analysis.SkroutzGreekStemmerBinderProcessor;
 
-public class SkroutzGreekStemmerPlugin extends Plugin {
+import org.elasticsearch.index.analysis.SkroutzGreekStemmerTokenFilterFactory;
 
-	@Override
-	public String description() {
-		return "Greek stemmer customized for the needs of www.skroutz.gr";
-	}
+import java.util.Map;
 
-	@Override
-	public String name() {
-		return "skroutz-greek-stemmer";
-	}
+import static java.util.Collections.singletonMap;
 
-	public void onModule(AnalysisModule module) {
-		module.addProcessor(new SkroutzGreekStemmerBinderProcessor());
-	}
+public class SkroutzGreekStemmerPlugin extends Plugin implements AnalysisPlugin {
 
+	@Override
+	public Map<String, AnalysisProvider<TokenFilterFactory>> getTokenFilters() {
```

Comment:
This might help:

[https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/plugins/AnalysisPlugin.java#L41](url)